app/services/vendor_api/generate_test_applications_for_provider.rb

```diff
@@ -12,16 +12,45 @@ def call(provider:, courses_per_application:, count:, for_ratified_courses: fals
 
   private
 
-    def course_list_for_provider(provider, for_ratified_courses)
+    def course_list_for_provider(provider, for_ratified_courses, mixed_application)
       if for_ratified_courses
-        GetCoursesRatifiedByProvider.call(provider: provider)
+        courses_ratified_by_provider(provider)
+      elsif mixed_application
+        [
+          courses_run_by_provider(provider).first,
+          courses_ratified_by_provider(provider).first,
+          course_run_by_test_provider,
+        ].compact
```

Comment:
Currently rethinking this...

Issue at the moment is that this won't work if the provider doesn't accredit anywhere. Maybe we can fall back to another course_run_by_provider in this case?

Reply:
One question is, do we want to make sure that all of the applications are "mixed", or just pick randomly from a mixed set of courses? The reason I didn't go for the latter is because asking for "mixed" suggests that you will definitely get a mixture. The former is slightly less nice to look at codewise though

Reply:
Agree that mixed should ensure returning at least one of each to avoid confusion